src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -587,13 +587,69 @@ public override void VisitFunctionCallSyntax(FunctionCallSyntax syntax)
                 {
                     case ErrorSymbol errorSymbol:
                         // function bind failure - pass the error along
-                        return new ErrorTypeSymbol(errors.Concat(errorSymbol.GetDiagnostics()));
+                        return new UnassignableTypeSymbol(errors.Concat(errorSymbol.GetDiagnostics()));
 
                     case FunctionSymbol function:
-                        return GetFunctionSymbolType(syntax, function, syntax.Arguments, argumentTypes, errors);
+                        return GetFunctionSymbolType(function, syntax.OpenParen, syntax.CloseParen, syntax.Arguments, argumentTypes, errors);
 
                     default:
-                        return new ErrorTypeSymbol(errors.Append(DiagnosticBuilder.ForPosition(syntax.Name.Span).SymbolicNameIsNotAFunction(syntax.Name.IdentifierName)));
+                        return new UnassignableTypeSymbol(errors.Append(DiagnosticBuilder.ForPosition(syntax.Name.Span).SymbolicNameIsNotAFunction(syntax.Name.IdentifierName)));
+                }
+            });
+
+        public override void VisitInstanceFunctionCallSyntax(InstanceFunctionCallSyntax syntax)
+            => AssignType(syntax, () => {
+
+                base.VisitInstanceFunctionCallSyntax(syntax);
+
+                var errors = new List<ErrorDiagnostic>();
+                
+                var baseType = typeManager.GetTypeInfo(syntax.BaseExpression);
+                CollectErrors(errors, baseType);
+
+                if (errors.Any())
+                {
+                    return new UnassignableTypeSymbol(errors);
+                }
+
+                var argumentTypes = syntax.Arguments.Select(arg => typeManager.GetTypeInfo(arg)).ToList();
+
+                foreach (TypeSymbol argumentType in argumentTypes)
+                {
+                    CollectErrors(errors, argumentType);
+                }
+
+                switch (bindings[syntax])
+                {
+                    case ErrorSymbol errorSymbol:
+                        // bind bind failure - pass the error along
+                        return new UnassignableTypeSymbol(errors.Concat(errorSymbol.GetDiagnostics()));
+
+                    case FunctionSymbol function:
+                        return GetFunctionSymbolType(function, syntax.OpenParen, syntax.CloseParen, syntax.Arguments, argumentTypes, errors);
+
+                    case VariableSymbol variable:
+                        // this is the case when a variable name uses the same name as the fully qualified function name
+                        // e.g. var resourceGroup = az.resourceGroup().name
```

Comment:
Why would the instance function call syntax be bound to `VariableSymbol` in this example? That doesn't seem correct.

Reply:
You are right, removed this case statement, not needed.